store/src/java/com/zimbra/cs/html/owasp/OwaspHtmlSanitizer.java

```diff
@@ -98,12 +98,12 @@ private String cleanMalformedHtml(String str) throws UnsupportedEncodingExceptio
             Tidy tidy = new Tidy();
             tidy.setInputEncoding("UTF-8");
             tidy.setOutputEncoding("UTF-8");
-            tidy.setPrintBodyOnly(true);
+            tidy.setPrintBodyOnly(false);
             tidy.setXHTML(true);
             ByteArrayInputStream inputStream = new ByteArrayInputStream(str.getBytes("UTF-8"));
             ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
             tidy.parseDOM(inputStream, outputStream);
-            String outStream = outputStream.toString("UTF-8");
+            String outStream = outputStream.toString("UTF-8").trim();
             if (outStream.isEmpty() || outStream == null) {
```

Comment:
1. `if (outStream.isEmpty() || outStream == null) {`  this is wrong check.  Null check does not make any sense.
if outStream is null then outStream.isEmpty() will throw NullPointerException. 

See below code snippet. it will always throw NPE. 
`        String b = null; 
        if(b.isEmpty() || b == null) {
            System.out.println("blank");
        }` 

The correct code first null check and then operations. 

`if (outStream == null || outStream.isEmpty()) {` 

2. you can use trim() while validating. do not unnecessary remove  leading and trailing spaces and set back to the string. 

`if (outStream == null || outStream.trim().isEmpty()) {`